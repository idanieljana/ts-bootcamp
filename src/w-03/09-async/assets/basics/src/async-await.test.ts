/* eslint-disable no-console */
/* eslint-disable @typescript-eslint/no-use-before-define */

describe('async/await', () => {
  describe.skip('await keyword place difference demo', () => {
    test('should wait 2 seconds and return sum', async () => {
      const result = await addTwoNumbersSlow();
      expect(result).toBeGreaterThan(0);
    });
    test('should wait 1 second and return sum', async () => {
      const result = await addTwoNumbersFaster();
      expect(result).toBeGreaterThan(0);
    });
  });
});

async function getOperandWithDelay() {
  return new Promise<number>((resolve) => {
    setTimeout(() => {
      resolve(Math.floor(Math.random() * Math.floor(10)));
    }, 1000);
  });
}

async function addTwoNumbersSlow() {
  console.time('Time elapsed');
  const operand1 = await getOperandWithDelay();
  const operand2 = await getOperandWithDelay();
  const sum = operand1 + operand2;
  console.timeEnd('Time elapsed');
  return sum;
}

async function addTwoNumbersFaster() {
  console.time('Time elapsed');
  const operandPromise1 = getOperandWithDelay();
  const operandPromise2 = getOperandWithDelay();
  const operand1 = await operandPromise1;
  const operand2 = await operandPromise2;
  const sum = operand1 + operand2;
  console.timeEnd('Time elapsed');
  return sum;
}
