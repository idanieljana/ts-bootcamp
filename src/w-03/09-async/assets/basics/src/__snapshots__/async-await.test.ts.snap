// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async/await should return the list of directors (version) should return list of directors 1`] = `
Array [
  Object {
    "id": 1,
    "movies": "http://localhost:3000/api/directors/1/movies",
    "name": "James Cameron",
  },
  Object {
    "id": 2,
    "movies": "http://localhost:3000/api/directors/2/movies",
    "name": "Quentin Tarantino",
  },
  Object {
    "id": 3,
    "movies": "http://localhost:3000/api/directors/3/movies",
    "name": "Wes Anderson",
  },
  Object {
    "id": 4,
    "movies": "http://localhost:3000/api/directors/4/movies",
    "name": "Stanley Kubrick",
  },
]
`;

exports[`async/await should return the list of directors (version) should return list of directors movies 1`] = `
Array [
  Array [
    Object {
      "id": 1,
      "reviews": "http://localhost:3000/api/movies/1/reviews",
      "title": "Titanic",
    },
    Object {
      "id": 2,
      "reviews": "http://localhost:3000/api/movies/2/reviews",
      "title": "The Terminator",
    },
    Object {
      "id": 3,
      "reviews": "http://localhost:3000/api/movies/3/reviews",
      "title": "Avatar",
    },
  ],
  Array [
    Object {
      "id": 4,
      "reviews": "http://localhost:3000/api/movies/4/reviews",
      "title": "Django Unchained",
    },
    Object {
      "id": 5,
      "reviews": "http://localhost:3000/api/movies/5/reviews",
      "title": "Inglourious Basterds",
    },
    Object {
      "id": 6,
      "reviews": "http://localhost:3000/api/movies/6/reviews",
      "title": "Grindhouse",
    },
  ],
  Array [
    Object {
      "id": 7,
      "reviews": "http://localhost:3000/api/movies/7/reviews",
      "title": "The Grand Budapest Hotel",
    },
    Object {
      "id": 8,
      "reviews": "http://localhost:3000/api/movies/8/reviews",
      "title": "Moonrise Kingdom",
    },
    Object {
      "id": 9,
      "reviews": "http://localhost:3000/api/movies/9/reviews",
      "title": "Fantastic Mr. Fox",
    },
  ],
  Array [
    Object {
      "id": 10,
      "reviews": "http://localhost:3000/api/movies/10/reviews",
      "title": "2001: A Space Odyssey",
    },
    Object {
      "id": 11,
      "reviews": "http://localhost:3000/api/movies/11/reviews",
      "title": "The Shining",
    },
    Object {
      "id": 12,
      "reviews": "http://localhost:3000/api/movies/12/reviews",
      "title": "A Clockwork Orange",
    },
  ],
]
`;
